mtp_api/apps/core/management/commands/dump_for_ap.py

```diff
@@ -68,25 +68,59 @@ def serialise(self, record):
 
 class CreditSerialiser(Serialiser, serialised_model=Credit):
     headers = Serialiser.headers + [
-        'Internal ID', 'URL',
-        'Date started', 'Date received', 'Date credited',
+        'Internal ID',
+        'URL',
+        'Date started',
+        'Date received',
+        'Date credited',
         'Amount',
-        'Prisoner number', 'Prisoner name', 'Prison',
-        'Sender name', 'Payment method',
-        'Bank transfer sort code', 'Bank transfer account', 'Bank transfer roll number',
-        'Debit card number', 'Debit card expiry', 'Debit card billing address',
-        'Sender email', 'Sender IP address',
+        'Prisoner number',
+        'Prisoner name',
+        'Prison',
+        'Owner username',
+        'Blocked',
+        'Sender name',
+        'Payment method',
+        'Bank transfer sort code',
+        'Bank transfer account',
+        'Bank transfer roll number',
+        'Debit card first six digits',
+        'Debit card last four digits',
+        'Debit card expiry',
+        'Debit card billing address line 1',
+        'Debit card billing address line 2',
+        'Debit card billing address city',
+        'Debit card billing address postcode',
+        'Debit card billing address country',
+        'Sender email',
+        'Sender IP address',
         'Status',
         'NOMIS transaction',
         'WorldPay order code',
+        'Security check status',
+        'Security check codes',
+        'Exported at',
     ]
 
+    exported_at_local_time = timezone.now()
+
     def serialise(self, record: Credit):
         status = record.status
         if status:
             status = CREDIT_STATUS.for_value(status).display
         else:
             status = 'Anonymous'
+
+        if hasattr(record, 'security_check'):
+            security_check_status = record.security_check.status
+            if len(record.security_check.rules) > 0:
```

Comment:
the `rules` field can be `None` so perhaps just `if record.security_check.rules:` is safer

Reply:
This causes the tests to fail locally - I believe because test data doesn't have any security rules checked:
`django.db.models.fields.related_descriptors.RelatedObjectDoesNotExist: Credit has no security_check.`

I would have hoped for a safe navigation operator (`&.` in ruby) but that isn't available in python yet it seems (hopefully coming):  https://www.python.org/dev/peps/pep-0505/

Any other ways to check, aside from `if record.security_check && record.security_check.rules:` (assuming python would exit the condition at the first statement)

Reply:
Gotcha - and after our chat and more testing I think the problem that did exist with test data has gone away following the JSON Lines conversion (should it have actually existed in the first place, as I'm now not so sure it did!)